lightning/src/chain/channelmonitor.rs

```diff
@@ -2041,8 +2052,23 @@ impl<Signer: WriteableEcdsaChannelSigner> ChannelMonitorImpl<Signer> {
 		// Prune HTLCs from the previous counterparty commitment tx so we don't generate failure/fulfill
 		// events for now-revoked/fulfilled HTLCs.
 		if let Some(txid) = self.prev_counterparty_commitment_txid.take() {
-			for &mut (_, ref mut source) in self.counterparty_claimable_outpoints.get_mut(&txid).unwrap() {
-				*source = None;
+			let cur_claimables = self.counterparty_claimable_outpoints.get(
+				&self.current_counterparty_commitment_txid.unwrap()).unwrap();
+			for (_, ref source_opt) in self.counterparty_claimable_outpoints.get(&txid).unwrap() {
+				if let Some(source) = source_opt {
+					let mut also_in_cur_commitment = false;
```

Comment:
```suggestion
					let also_in_cur_commitment = cur_claimables.iter().any(|(_, s)| s == source_opt);
```

Reply:
Alternatively you can do
```rust
let cur_claimables = cur_claimables.iter().map(|(_, s)| s).flatten().collect::<HashSet<_>>();
```
and then use here:
```rust
let also_in_cur_commitment = cur_claimables.contains(source);
```
This will make the search `O(N)` instead of `O(N^2)` at the expense of higher memory use and the fixed cost of the hashset construction